### PR TITLE
adds docs around the uris concept

### PR DESCRIPTION
**Pull request by mhausenblas**
Currently, the doc lacks explanation of the `uris` concept. This PR extends the docs to explain how `uris` can be used.

### CODE REVIEW

**Comment by air** [comment]
Mesos Fetcher: Knowing about this mechanism is important. Linking to the fetcher.cpp is pretty hardcore though for a Hello World doc. Is there an easier explanation somewhere?

**Comment by air** [comment]
On the first mention of sandbox, we need to link out to an explanation of what that is. A quick search did not turn up any good options.

**Comment by air** [comment]
Overall this is great. Ideas for the end of that section:

- A typical pattern is to have your automated build system place the app binary in a location that's downloadable via URI.
- We should point out that URIs can use protocols like hdfs: - I believe that's another common pattern, but someone with more Mesos experience should confirm.

**Comment by drexin** [comment]
@mhausenblas Could you please quash this into a single commit?

File: docs/docs/deployments.md [hunk 1]
```diff
@@ -18,6 +18,97 @@ Multiple deployments can be performed at the same, as long as one application is
...
+    "id": "basic-0", 
+    "cmd": "while [ true ] ; do echo 'Hello Marathon' ; sleep 5 ; done",
+    "cpus": 0.1,
+    "mem": 10.0,
+    "instances": 1
+}
+```
+
+Note that `cmd` in the above example is the command that gets executed. 
+Its value is wrapped by the underlying [Mesos fetcher](https://github.com/apache/mesos/blob/master/src/launcher/fetcher.cpp) via `/bin/sh -c ${cmd}`.
```
Comment (drexin) [inline docs/docs/deployments.md]: The command is actually being wrapped by the executor, the fetcher only fetches the resources required to execute the command.

Comment (drexin) [inline docs/docs/deployments.md]: Also I don't think we need such a reference here.

File: docs/docs/deployments.md [hunk 2]
```diff
@@ -18,6 +18,97 @@ Multiple deployments can be performed at the same, as long as one application is
...
+
+<p class="text-center">
+  <img src="{{ site.baseurl}}/img/marathon-basic-0.png" width="800" height="573" alt="Marathon deployment example: simple bash commands">
+</p>
+
+What happens here is that Marathon builds on Mesos' sandbox concept: a special directory on each slave that acts as the execution environment (from a storage perspective) and also contains relevant log files as well as `stderr` and `stdout` for the command being executed. See also the role of the sandbox in [debugging distributed apps](https://docs.mesosphere.com/tutorials/debugging-a-mesosphere-cluster/).
```
Comment (drexin) [inline docs/docs/deployments.md]: Marathon doesn't really build on the sandbox concept, it doesn't even know about it. We only send Mesos the TaskInfo and how Mesos handles it is really transparent to us.

File: docs/docs/deployments.md [hunk 3]
```diff
@@ -18,6 +18,99 @@ Multiple deployments can be performed at the same, as long as one application is
...
+The sandbox is a special directory on each slave that acts as the execution environment (from a storage perspective) and also contains relevant log files as well as `stderr` and `stdout` for the command being executed. See also the role of the sandbox in [debugging distributed apps](https://docs.mesosphere.com/tutorials/debugging-a-mesosphere-cluster/).
+
+For any non-trivial deployment you typically depend on a collection of resources, that is, files and/or archives of files. To deal with this, Marathon has the concept of `uris`. It leverages the Mesos fetcher to do the legwork in terms of downloading (and potentially) extracting resources.
+
+But before we dive into this topic, let's have a look at an example:
+
+```json
+{
+    "id": "basic-1", 
+    "cmd": "cool-script.sh",
```
Comment (ConnorDoyle) [inline docs/docs/deployments.md]: Probably, pwd is not on the PATH: `./cool-script.sh`.  Also, have you tried this?  It looks like in Mesos `0.22.0` you'd have to do `chmod ug+x cool-script.sh && ./cool-script.sh`.

Comment (mhausenblas) [inline docs/docs/deployments.md]: IMO: if we do this, we should do it properly. My plan was to add cool-script.sh to the doc/ dir here (a well-known) location and then also add a screen shot. Not sure how deep we should go concerning concrete Mesos versions (note: I'm using latest playa to test).

Comment (mhausenblas) [inline docs/docs/deployments.md]: Resolved with adding a note explaining the difference between pre and post Mesos 0.21

Comment (ConnorDoyle) [inline docs/docs/deployments.md]: This command is still missing the `./`.

**Comment by ConnorDoyle** [comment]
LGTM except for the last outstanding comment.  Thanks @mhausenblas!

**Comment by mhausenblas** [comment]
OK, all comments addressed, should be good to ship now @ConnorDoyle

**Comment by ConnorDoyle** [comment]
Syncing the docs site branch...

**Comment by ssk2** [comment]
Just realised that this is on the deployments page. I don't think this is the right place for docs about launching an app with URIs - in fact, it'd probably be useful to have a specific "URIs" page (similar to https://mesosphere.github.io/marathon/docs/task-environment-vars.html). What do you think @mhausenblas?

**Comment by mhausenblas** [comment]
Thanks for the feedback, @ssk2 — from my POV (trying to represent the 'average' dev) this was the natural place, that is, I would have expected to find it exactly there. Having said this, I do agree that it would deserve its own page (esp. given the upcoming stuff, more screenshots, etc.). The main challenge we have, from a docs user perspective, is that there are already (too) many menu items on the left-hand side. So, I'm unsure about adding yet another menu item. Bottom-line: yes to create a new, dedicated page, unsure how to organize it. @ConnorDoyle anything to add?

**Comment by ssk2** [comment]
Hm, what you've contributed looks more like a tutorial of the basics of an application definition and is not necessarily related to deployments. On second thought, a URI page would be the wrong place to put it - I think it'd go well as a new guide, "App Definition Basics"?

**Comment by mhausenblas** [comment]
I'm sorry but now you've lost me @ssk2 . What if not deployment is this? I've never seen the term 'application definition' (where is this from?) and wouldn't know the difference to deployment either. Can you clarify pls?

**Comment by ConnorDoyle** [comment]
In Marathon (internals at least) an App definition is the description of your app that you send to Marathon over HTTP.  A deployment is a process, consisting of a series of steps, each step consisting of one or more actions.  Deployment steps are things like "Start Application", "Restart Application", "Scale Up Application", etc.

**Comment by mhausenblas** [comment]
Thank you for the clarification @ConnorDoyle — that makes sense. With this information now I'd suggest a totally different approach: let's not think about how *we* organize stuff internally, but how devops are using it, since devops are also only just users and UX should always have highest priority. I don't know how we wanna do that but happy to jump on a planning/brainstorming G+ hangout. Any ideas, @ConnorDoyle or @ssk2?

**Comment by ssk2** [comment]
Let's work with @joel-hamill and @zedshaw to make sure our approach to docs is consistent. We may want to re-architect these at the same time as bringing Chronos docs up to parity. In the meantime, I'm going to propose an alternative in a PR. Will tag you for comment shortly.

**Comment by air** [comment]
Since this is hello world stuff, it should be somewhere on the Getting
Started path.

Currently the Getting Started page seems to cover installation but has
nothing about running something simple?

**Pull request merged**